src/s3000_laser.cc

```diff
@@ -129,62 +138,61 @@ class s3000node {
 
   void ConnectTest(diagnostic_updater::DiagnosticStatusWrapper& status)
   {
-    laser->Open();
-    status.summary(0, "Connected successfully.");
+    if (laser->Open() == 0)
+      status.summary(0, "Connected successfully.");
```

Comment:
Use the enums from `diagnostic_msgs::DiagnosticStatus`, e.g.: `diagnostic_msgs::DiagnosticStatus::ERROR`